### PR TITLE
feat(sns): SNS Governance upgrades its Swap

### PR DESCRIPTION
**Pull request by aterga**
This PR makes SNS Governance actually upgrade its Swap. This is a step towards making Swaps proper SNS canisters.

< [Previous PR](https://github.com/dfinity/ic/pull/2286) |

### CODE REVIEW

File: rs/nervous_system/integration_tests/tests/sns_release_qualification.rs
Comment (max-dfinity) [inline rs/nervous_system/integration_tests/tests/sns_release_qualification.rs:285]: Are you going to change the existing tests and remove this test eventually?  I thought we discussed that yesterday, since the other tests could theoretically already handle this with some slight modifications.

I'd make a TODO note if that's being blocked here.

**Review by max-dfinity — APPROVED** [review]

File: rs/nervous_system/integration_tests/tests/sns_release_qualification.rs
Comment (aterga) [reply rs/nervous_system/integration_tests/tests/sns_release_qualification.rs:295]: Added a DO NOT MERGE and an explanation to the comment above this test.

File: rs/nervous_system/integration_tests/tests/sns_release_qualification.rs
Comment (basvandijk) [inline rs/nervous_system/integration_tests/tests/sns_release_qualification.rs:306]: ?

**Review by basvandijk — APPROVED** [review]

File: rs/nervous_system/integration_tests/tests/sns_release_qualification.rs
Comment (aterga) [reply rs/nervous_system/integration_tests/tests/sns_release_qualification.rs:306]: This test should be removed before we merge this PR, as the normal qualification tests would cover this scenario. 

For context, this PR is blocked on https://github.com/dfinity/ic/pull/2427

**Review by anchpop — APPROVED** [review]

**Review by max-dfinity — APPROVED** [review]

**Pull request merged**
